trezorlib/tx_api.py

```diff
@@ -97,6 +112,13 @@ def get_tx(self, txhash):
             o = t._add_bin_outputs()
             o.amount = int(Decimal(vout['value']) * 100000000)
             o.script_pubkey = binascii.unhexlify(vout['scriptPubKey']['hex'])
+            if(self.bip115 and len(o.script_pubkey) == 63 and o.script_pubkey[-1:] == b'\xb4'):
+                o.block_hash = o.script_pubkey[-37:-5]
+                o.block_height = int.from_bytes(o.script_pubkey[-4:-1], byteorder='little')
```

Comment:
again, what is this magic exactly?

1. don't wrap the if-condition in parentheses
2. to check for the last byte of scriptPubKey, do this: `.script_pubkey[-1] == 0xb4`
3. please add a comment explaining the numbers. perhaps a link to some spec would be helpful. ideally also describe the structure you're accessing.
4. is there a reason to count backwards? you are enforcing a fixed length of the script anyway.

Reply:
just noticed now: this extraction of `block_hash` and `block_height` only works with `o.script_pubkey`.

That means that it doesn't need to be done here. Instead it should be done in Trezor itself, and the new fields should be removed from `InputType`